### PR TITLE
Feat/update v0.4.1

### PR DESCRIPTION
**Pull request by MementoRC**
Mainly add support for the new features of SECP256K1.

   - Replaced deprecated functions.
   - Moved project description to TOML (is that ok?)
   - Added tox test for python 3.12
   - 
The .h are created automatically from the sec256pk1 with gcc -E (and a few tricks) so it formats a little oddly, creating lots of changes (like int* a -> int *a) which may make the review tedious
In the conda recipe (and the github repo for libsecp256k1-py-bindings), the build.py was modified to make it more automated and able to add either dynamic or static Clib. The compilation also uses an ABI version of CFFI lib

### CODE REVIEW

**Comment by ofek** [comment]
Let's update:

- https://github.com/ofek/coincurve/blob/2dfea673835e524ffaeb11dc1c2aecaf23188195/.github/workflows/build.yml#L41 to `3.12`
- Every `pypa/cibuildwheel@v2.11.2` to `pypa/cibuildwheel@v2.16.2`

**Comment by MementoRC** [comment]
@ofek, Oh, we were on the same wavelength. I broke the conda-forge recipe though, I am trying to fix it

**Comment by ofek** [comment]
Is this ready for review/merge?

**Comment by MementoRC** [comment]
@ofek, yes, it passes the conda recipe checks as well

**Comment by MementoRC** [comment]
@ofek Are there channges needed?

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,53 @@
+[build-system]
+requires = ["setuptools>=61", "cffi>=1.3.0", "urllib3>2"]
```
Comment (ofek) [inline pyproject.toml:2]:
```suggestion
requires = ["setuptools>=61", "cffi>=1.3.0"]
```

**Review by ofek — CHANGES_REQUESTED** [review]

File: setup.py [hunk 2]
```diff
@@ -232,84 +245,101 @@ def run(self):
...
 
-    class Distribution(_Distribution):
-        def is_pure(self):
-            return False
+class BuildCFFIForSharedLib(_build_ext):
+    def build_extensions(self):
+        build_script = os.path.join('_cffi_build', 'build_shared.py')
+        c_file = self.extensions[0].sources[0]
+        python_exe = shutil.which('python', path=os.environ['PATH'])
+        subprocess.run([python_exe, build_script, c_file, '0'], shell=False, check=True)  # noqa S603
```
Comment (ofek) [inline setup.py:254]:
```suggestion
        subprocess.run([sys.executable, build_script, c_file, '0'], shell=False, check=True)  # noqa S603
```

**Review by ofek — CHANGES_REQUESTED** [review]

**Review by ofek — APPROVED** [review]
Thank you very much!

**Comment by ofek** [comment]
https://github.com/ofek/coincurve/releases/tag/v19.0.0

**Pull request merged**